sql/database.go

```diff
@@ -204,46 +188,22 @@ func exec(conn *sqlite.Conn, query string, encoder Encoder, decoder Decoder) (in
 	}
 }
 
-// Tx is wrapper for database transaction.
-type Tx struct {
-	db        *Database
-	conn      *sqlite.Conn
-	committed bool
-	err       error
-}
-
-func (tx *Tx) begin() error {
-	stmt := tx.conn.Prep("BEGIN;")
-	_, err := stmt.Step()
-	if err != nil {
-		return fmt.Errorf("begin: %w", err)
+// GetSQLiteError returns sqlite error for the given error.
+func (db *Database) GetSQLiteError(err error) *sqlite.Error {
```

Comment:
is it possible to use errors.As? why?

Reply:
sqllite not use go's original error. They have their own struct, which just implement `error` interface. When they call `Error() string` - it include also issue with sql, which can be different any time.

so need deep unwrap to get this object and get code.

Reply:
```
testErr := errors.Wrap(errors.Wrap(sqlite.Error{
  Code:  1,
  Query: "123",
  Msg:   "234",
}, "wrapped sqlite error"), "wrapped error")
resErr := sqlite.Error{Code: 2, Query: "1234", Msg: "234"}
r := errors.As(testErr, &resErr) // true
```
this code will return true even if sqlCode is different